pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -796,108 +801,72 @@ func createHostDnsmasqFile(ctx *zedrouterContext, bridge string) {
 	}
 }
 
-// Returns an IP address as a string, or "" if not found.
-func lookupOrAllocateIPv4(
-	ctx *zedrouterContext,
-	status *types.NetworkInstanceStatus,
-	mac net.HardwareAddr) (string, error) {
-
+// Returns an IP address as a string, or "" on error
+func lookupOrAllocateIPv4(netInstStatus *types.NetworkInstanceStatus,
+	appID string, appNum int, mac net.HardwareAddr) (string, error) {
 	log.Functionf("lookupOrAllocateIPv4(%s-%s): mac:%s\n",
-		status.DisplayName, status.Key(), mac.String())
-	// Lookup to see if it exists
-	if ip, ok := status.IPAssignments[mac.String()]; ok {
+		netInstStatus.DisplayName, netInstStatus.Key(), mac.String())
+	networkID := netInstStatus.Key()
+	ipAddrMapInfo := ipAddrMapGet(netInstStatus)
+	appNumMapInfo := uNetAppNumMapGet(networkID)
+	if appNum == types.AppNumInvalid || ipAddrMapInfo == nil {
+		log.Functionf("(%s)no ip address", mac.String())
+		return "", nil
```

Comment:
Why isn't this an error? You failed to allocate an appnum hence you can't get an IP address.